src/resources/checkout-intents.ts

```diff
@@ -96,6 +98,262 @@ export class CheckoutIntentsResource extends APIResource {
+   *   console.log('Order failed:', checkoutIntent.failureReason);
+   * }
+   * ```
+   */
+  async pollUntilCompleted(
+    id: string,
+    options?: RequestOptions & { pollIntervalMs?: number; maxAttempts?: number },
+  ): Promise<CheckoutIntent> {
+    return this.pollUntil(id, (intent) => intent.state === 'completed' || intent.state === 'failed', options);
+  }
+
+  /**
+   * A helper to poll a checkout intent until it's ready for confirmation
+   * (awaiting_confirmation state) or has failed. This is typically used after
+   * creating a checkout intent to wait for the offer to be retrieved from the merchant.
+   *
+   * The intent can reach awaiting_confirmation (success - ready to confirm) or failed
+   * (offer retrieval failed). Always check the state after polling.
+   *
+   * @example
+   * ```ts
+   * const intent = await client.checkoutIntents.pollUntilAwaitingConfirmation('id');
+   *
+   * if (intent.state === 'awaiting_confirmation') {
+   *   // Review the offer before confirming
+   *   console.log('Total:', intent.offer.cost.total);
+   * } else if (intent.state === 'failed') {
+   *   // Handle failure (e.g., offer retrieval failed, product out of stock)
+   *   console.log('Failed:', intent.failureReason);
+   * }
+   * ```
+   */
+  async pollUntilAwaitingConfirmation(
+    id: string,
+    options?: RequestOptions & { pollIntervalMs?: number; maxAttempts?: number },
+  ): Promise<CheckoutIntent> {
+    return this.pollUntil(
+      id,
+      (intent) => intent.state === 'awaiting_confirmation' || intent.state === 'failed',
+      options,
+    );
+  }
+
+  /**
+   * A helper to create a checkout intent and poll until it's ready for confirmation.
+   * This follows the Rye documented flow: create → poll until awaiting_confirmation.
+   *
+   * After this method completes, you should review the offer (pricing, shipping, taxes)
+   * with the user before calling confirm().
+   *
+   * @example
+   * ```ts
+   * // Phase 1: Create and wait for offer
+   * const intent = await client.checkoutIntents.createAndPoll({
+   *   buyer: {
+   *     address1: '123 Main St',
+   *     city: 'New York',
+   *     country: 'United States',
+   *     email: 'john.doe@example.com',
+   *     firstName: 'John',
+   *     lastName: 'Doe',
+   *     phone: '+1234567890',
+   *     postalCode: '10001',
+   *     province: 'NY',
+   *   },
+   *   productUrl: 'https://example.com/product',
+   *   quantity: 1,
+   * });
+   *
+   * // Review the offer with the user
+   * console.log('Total:', intent.offer.cost.total);
+   *
+   * // Phase 2: Confirm with payment
+   * const completed = await client.checkoutIntents.confirmAndPoll(intent.id, {
+   *   paymentMethod: { type: 'stripe_token', stripeToken: 'tok_visa' }
+   * });
+   * ```
+   */
+  async createAndPoll(
+    body: CheckoutIntentCreateParams,
+    options?: RequestOptions & { pollIntervalMs?: number; maxAttempts?: number },
+  ): Promise<CheckoutIntent> {
```

Comment:
would be great to return a narrower type here; we only ever return checkout intents in failed or completed state from this function

ditto other helpers (e.g. createAndPoll only ever returns CIs in awaiting_confirmation or failed state)